src/evict/evict_lru.c

```diff
@@ -1443,19 +1443,82 @@ __wt_evict_lru_page(WT_SESSION_IMPL *session, int is_server)
 }
 
 /*
- * __wt_cache_wait --
- *	Wait for space in the cache.
+ * __cache_check --
+ *	Return the cache percent-full, and if eviction targets need addressing.
+ */
+static inline void
+__cache_check(WT_CONNECTION_IMPL *conn, int *pctp, int *targetp)
```

Comment:
Thank you -- I meant to go back and look at that function and forgot. I've cleaned it up and simplified it, there's no need for the `targetp` argument at all, the function can return true/false if the application thread should be involved in eviction.